utils.py

```diff
@@ -109,7 +109,14 @@ def json_minify(data: JsonType | list[JsonType]) -> str:
 
 
 def timestamp(string: str) -> datetime:
+  try:
     return datetime.strptime(string, "%Y-%m-%dT%H:%M:%SZ").replace(tzinfo=timezone.utc)
+  except ValueError:
+    try:
+      return datetime.strptime(string, "%Y-%m-%dT%H:%M:%S.%fZ").replace(tzinfo=timezone.utc)
+    except ValueError:
+      return
+
```

Comment:
```suggestion
    try:
        return datetime.strptime(string, "%Y-%m-%dT%H:%M:%S.%fZ").replace(tzinfo=timezone.utc)
    except ValueError:
        return datetime.strptime(string, "%Y-%m-%dT%H:%M:%SZ").replace(tzinfo=timezone.utc)
```

Reply:
4-space indent, no extra empty line at the end, and no undefined return value - the second `return` has to succeed, otherwise the resulting `ValueError` has to "bubble-up" as a normal exception.

Reply:
Makes sense, thank you. I'm fairly new to Python so I wasn't sure if I needed an additional except or all cases that didn't match.